src/app/clusters/ota-provider/ota-provider.cpp

```diff
@@ -131,22 +134,22 @@ bool emberAfOtaSoftwareUpdateProviderClusterNotifyUpdateAppliedCallback(
     ChipLogDetail(Zcl, "  Update Token: %zu", commandData.updateToken.size());
     ChipLogDetail(Zcl, "  Software Version: %" PRIu32, commandData.softwareVersion);
 
-    if (SendStatusIfDelegateNull(endpoint))
+    if (SendStatusIfDelegateNull(commandObj, commandPath))
     {
         return true;
     }
 
     if (updateToken.size() > kUpdateTokenMaxLength || updateToken.size() < kUpdateTokenMinLength)
     {
         ChipLogError(Zcl, "expected size %zu for UpdateToken, got %zu", kUpdateTokenMaxLength, updateToken.size());
-        emberAfSendImmediateDefaultResponse(EMBER_ZCL_STATUS_INVALID_COMMAND);
+        commandObj->AddStatus(commandPath, Status::InvalidCommand);
         return true;
     }
 
     status = delegate->HandleNotifyUpdateApplied(commandObj, commandPath, commandData);
```

Comment:
Perhaps out of the scope of this PR but [emberAfSendImmediateDefaultResponse](https://github.com/project-chip/connectedhomeip/blob/master/examples/ota-provider-app/ota-provider-common/OTAProviderExample.cpp#L341) is still being called in this function

Reply:
Out of scope, yes, but we should sort that out too.  I filed https://github.com/project-chip/connectedhomeip/issues/15889 on getting clarity on who is responsible for what and doing that.